.storybook/main.js

```diff
@@ -32,4 +35,18 @@ module.exports = {
       },
     },
   ],
+  viteFinal: (config) => {
+    const pluginsWithoutReact = config.plugins.filter(plugin => !Array.isArray(plugin));
```

Comment:
Why we use here this isArray condition? Is there any documentation that all react plugins are placed in one array?

Reply:
I didn't find any official docs about it, but if you do a console.log of the plugins it return something like this:

```js
[
    {
          // vue-plugin
    },
    [
         {
             // react-plugin
         },
         // ...
    ],
    {
          // vue-plugin
    },
    // ...
]
```

That's why with isArray I can remove all unnecessary React plugins.

Reply:
Okay, let's add comment here why isArray is using to determine React plugins and use mergeConfig function in return statement.

Reply:
I believe this PR applies the workaround from this issue: https://github.com/storybookjs/storybook/issues/21681. I also find the solution easy to break if any other array is added or the structure of the plugins changes, we may break this fix and we will not notice that it no longer works or breaks new stuff. I'd rewrite this a bit like this:
```
    const reactPluginNameRegex = /^vite:react-/;
    const reactPluginsArrayIndex = config.plugins.findIndex(plugin => Array.isArray(plugin) && plugin.every(p => p.name.match(reactPluginNameRegex)));

    if (reactPluginsArrayIndex === -1) {
      throw new Error('Unable to patch plugins - cannot find vite:react plugins. Is this fix still necessary?');
    }

    // remove react plugins and replace them with vue plugins
    // see https://github.com/storybookjs/storybook/issues/21681
    // remove after migrating to SB 7
    config.plugins.splice(reactPluginsArrayIndex, 1, vueJsx());
    return config;
```

^^ This find the plugins by name, adds an exception when unable to find them. The issue also states that the problem has been removed with SB v7, I'd add a comment to remove it when migrating. WDYT?

Reply:
Yep, that was my thought too. As I can understand only React plugins are in array, but anyway your solution is more sustainable. Also, I guess we avoid this problem and remove this fix once we are on 7 version.

Actually, I tried to migrate chameleon to storybook 7, jsx works with no hack, but I needed to make some adjustments. So we could stick with this fix for now, and have a task to do for later time to migrate to storybook 7.

Reply:
yes, this approach is better than mine because I didn't take into account the possibility of another array being added in the plugins :sweat_smile:, Should I implement this instead?

Reply:
We can quickly spawn an empty project with SB 7 and check the stories there if they work with JSX, and how the plugins array looks like there. Changelog for SB 7 says they rewritten the vite support and framework structure, introduced this new preset (vue3-vite: https://github.com/storybookjs/storybook/blob/next/code/frameworks/vue3-vite/src/preset.ts), so it's very likely it has been fixed.

Reply:
Yes, jsx fixed in 7 version.

Reply:
I had a look another time and I can see that I used the same hack to filter react plugins for jsx :).